frontend/src/metabase/redux/downloads.ts

```diff
@@ -37,9 +39,76 @@ interface DownloadQueryResultsParams {
   params?: URLSearchParams | string;
 }
 
+export type ResourceType = "question" | "dashcard" | "ad-hoc-question";
+export type ResourceAccessedVia =
+  | "internal"
+  | "public-link"
+  | "static-embed"
+  | "interactive-iframe-embed"
+  | "sdk-embed";
+
+export type DownloadedResourceInfo = {
+  resourceType: ResourceType;
+  accessedVia: ResourceAccessedVia;
+};
+
+const getDownloadedResourceType = ({
+  dashboardId,
+  dashcardId,
+  uuid,
+  token,
+  question,
+}: Partial<DownloadQueryResultsOpts>): DownloadedResourceInfo => {
+  const cardId = question?.id();
+
+  const isInIframe = isWithinIframe();
+
+  if (dashcardId != null && token != null) {
+    return { resourceType: "dashcard", accessedVia: "static-embed" };
+  }
+
+  if (dashboardId != null && uuid != null) {
+    return { resourceType: "dashcard", accessedVia: "public-link" };
+  }
+
+  if (dashboardId != null && dashcardId != null) {
+    return {
+      resourceType: "dashcard",
+      accessedVia: isInIframe ? "interactive-iframe-embed" : "internal",
+    };
+  }
+
+  if (uuid != null) {
+    return { resourceType: "question", accessedVia: "public-link" };
+  }
+
+  if (token != null) {
+    return { resourceType: "question", accessedVia: "static-embed" };
+  }
+
+  if (cardId != null) {
+    return {
+      resourceType: "question",
+      accessedVia: isInIframe ? "interactive-iframe-embed" : "internal",
+    };
+  }
+
+  return {
```

Comment:
That probably mean, the code here was wrong (see `otherwise`) since it's in `EmbedFrame` which is used by `PublicOrEmbeddedDashboard

https://github.com/metabase/metabase/blob/292c6144036a86a53917f43e8e719a0140f92e69/frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.tsx#L161-L164

Reply:
@WiNloSt why is that wrong?
Embed frame is only used by static/public/sdk so if it's not the first two, it means it's the sdk right?

Reply:
For the first comment, as I'm OOO from this afternoon we could do something quick like
```
  return {
    resourceType: "ad-hoc-question",
    accessedVia: process.env.EMBEDDING_SDK_VERSION ? "sdk-embed" : isInIframe ? "interactive-iframe-embed" : "internal",
  };
```
I tested that it works, but this is getting out of hand, I'll write in the team channel about a proposal on how to handle the `accessed_via` in a more scalable way today

Reply:
I'm referring to Denis' [comment here](https://github.com/metabase/metabase/pull/46335#discussion_r1700996545)



> Should we add a check for SDK ? In scope of editable dashboard we will use a regular `Dashboard` component, and not `PublicOrEmbeddedDashboard`.

I understand that we don't use `publicOrEmbeddedDashboard` in the SDK.

Reply:
Thanks for the check! Scratch what I wrote then.

Reply:
So we were actually missing the value of accessed_via: embed-sdk for the analytics so I added it with a quick fix in b3c3afc34ea1d3ddb2fb41519ce6116996e6c084
I think it can work for now but I'd like to have a cleaner solution eventually, I'll post on the team channel later with a proposed solutions with contexts